src/odemis/util/dataio.py

```diff
@@ -309,6 +310,14 @@ def leader_quality(da):
 
         # Make sure they are not DataArrayShadow, because we need the data to do the stitching
         das = [da.getData() if isinstance(da, model.DataArrayShadow) else da for da in das]
+        # For now the stitching doesn't handle more than 2 dimensions, so force the data to be 2D
+        das_2d = []
+        for da in das:
+            try:
+                das_2d.append(img.ensure2DImage(da))
+            except ValueError:
+                logging.info("Skipping %s because it is not 2D", da)
+        das = das_2d
```

Comment:
So if there are no 2D data arrays this will just be an empty list, is that the expected behavior?

Reply:
Good remark... Actually there is code above that does already some filtering on the das, so this could have already happened. For now in such case, it fails with IndexError. We could catch this case and make it a little bit more understandable.

Honestly this function is not super robust. We probably should also check that all the files have the same amount of streams, as otherwise it's also going to go bad...